OConnor/resources/views/Elispot_Matrix.html

```diff
@@ -44,8 +46,12 @@
 		
 	if(matrixExps != undefined || tradExps != undefined){
 		document.getElementById('matches').innerHTML = 'Possible Matches';	
-		document.getElementById('experiments').innerHTML = '&nbsp Trad. 1: ' + '<a href=javascript:toExperiment(trad1)>' + trad1 + '</a>' + '&nbsp Trad. 2: ' + '<a href=javascript:toExperiment(trad2)>' + trad2 + '</a>' + '&nbsp Matrix 1: ' + '<a href=javascript:toExperiment(matrix1)>' + matrix1 + '</a>' + '&nbsp Matrix 2: ' + '<a href=javascript:toExperiment(matrix2)>' + matrix2 + '</a>';
-	}
+		document.getElementById('experiments').innerHTML = '&nbsp Trad. 1: ' + '<a id="trad1">' + trad1 + '</a>' + '&nbsp Trad. 2: ' + '<a id="trad2">' + trad2 + '</a>' + '&nbsp Matrix 1: ' + '<a id="matrix1">' + matrix1 + '</a>' + '&nbsp Matrix 2: ' + '<a id="matrix2">' + matrix2 + '</a>';
```

Comment:
I know this predates your change, but we need to encode before reflecting back into the HTML.

```suggestion
		document.getElementById('experiments').innerHTML = '&nbsp Trad. 1: ' + '<a id="trad1">' + LABKEY.Utils.encodeHtml(trad1) + '</a>' + '&nbsp Trad. 2: ' + '<a id="trad2">' + LABKEY.Utils.encodeHtml(trad2) + '</a>' + '&nbsp Matrix 1: ' + '<a id="matrix1">' + LABKEY.Utils.encodeHtml(matrix1) + '</a>' + '&nbsp Matrix 2: ' + '<a id="matrix2">' + LABKEY.Utils.encodeHtml(matrix2) + '</a>';
```

I think the rest of the code in this file is safe.

Reply:
But thanks to strict CSP, we don't have to encode anything anymore ;)